dom/src/DOMSource.ts

```diff
@@ -1,13 +1,23 @@
 import {Stream, MemoryStream} from 'xstream';
 import {PreventDefaultOpt} from './fromEvent';
+import {MainDOMSource} from './MainDOMSource';
+import {DocumentDOMSource} from './DocumentDOMSource';
+import {BodyDOMSource} from './BodyDOMSource';
+import {MockedDOMSource} from './mockDOMSource';
 export interface EventsFnOptions {
   useCapture?: boolean;
   preventDefault?: PreventDefaultOpt;
 }
 
-export interface DOMSource {
+export type DOMSource =
+  | MainDOMSource
+  | DocumentDOMSource
+  | BodyDOMSource
+  | MockedDOMSource;
```

Comment:
Is the union type here necessary to achieve this PR? I was intending to eventually place JSDocs in the methods of the `DOMSource` interface, so that docs for `elements`, `events`, `select` etc would show when hovering in VSCode or elsewhere in other IDEs. Currently, docs for these methods are dumped into the JSDoc for `makeDOMDriver` in `index.ts`.

With a union type, it's unfortunate we can't provide the JSDoc for each method, and people will be importing `import {DOMSource} from '@cycle/dom'` instead of `import {MainDOMSource} from '@cycle/dom'`

Reply:
AFAIK it is needed for the type inference to work. But I have to check this

Reply:
Was not needed, removed the union